src/ReverseProxy/Forwarder/ProtocolHelper.cs

```diff
@@ -28,43 +29,10 @@ public static string GetVersionPolicy(HttpVersionPolicy policy)
         };
     }
 
-    // NOTE: When https://github.com/dotnet/aspnetcore/issues/21265 is addressed,
-    // this can be replaced with `MediaTypeHeaderValue.IsSubsetOf(...)`.
     /// <summary>
     /// Checks whether the provided content type header value represents a gRPC request.
-    /// Takes inspiration from
-    /// <see href="https://github.com/grpc/grpc-dotnet/blob/3ce9b104524a4929f5014c13cd99ba9a1c2431d4/src/Shared/CommonGrpcProtocolHelpers.cs#L26"/>.
     /// </summary>
-    public static bool IsGrpcContentType(string? contentType)
-    {
-        if (contentType is null)
-        {
-            return false;
-        }
-
-        if (!contentType.StartsWith(GrpcContentType, StringComparison.OrdinalIgnoreCase))
-        {
-            return false;
-        }
-
-        if (contentType.Length == GrpcContentType.Length)
-        {
-            // Exact match
-            return true;
-        }
-
-        // Support variations on the content-type (e.g. +proto, +json)
-        var nextChar = contentType[GrpcContentType.Length];
-        if (nextChar == ';')
-        {
-            return true;
-        }
-        if (nextChar == '+')
-        {
-            // Accept any message format. Marshaller could be set to support third-party formats
-            return true;
-        }
-
-        return false;
-    }
+    public static bool IsGrpcContentType(string? contentType) =>
+        MediaTypeHeaderValue.TryParse(contentType, out var mediaType)
+        && mediaType.MatchesMediaType(GrpcContentType);
```

Comment:
Can we keep the `StartsWith` to avoid allocating the `MediaTypeHeaderValue` in the common case? 😃

```suggestion
        contentType is not null &&
        && contentType.StartsWith(GrpcContentType, StringComparison.OrdinalIgnoreCase)
        && MediaTypeHeaderValue.TryParse(contentType, out var mediaType)
        && mediaType.MatchesMediaType(GrpcContentType);
```